test/test_torch.py

```diff
@@ -6473,6 +6473,25 @@ def test_isfinite_isinf_isnan_complex(self, device, dtype):
         self.compare_with_numpy(torch.isinf, np.isinf, vals, device, dtype)
         self.compare_with_numpy(torch.isnan, np.isnan, vals, device, dtype)
 
+    @unittest.skipIf(not TEST_NUMPY, 'NumPy not found')
+    @dtypes(torch.complex64, torch.complex128)
+    def test_isreal_complex(self, device, dtype):
+        vals = (1, 1 + 1j, 2 + 0j, 3j, 2 - 1j, 2 - 0j)
+        self.compare_with_numpy(torch.isreal, np.isreal, vals, device, dtype)
+
+    @unittest.skipIf(not TEST_NUMPY, 'NumPy not found')
+    @dtypes(*torch.testing.get_all_dtypes())
+    def test_isreal_noncomplex(self, device, dtype):
+        if dtype is not torch.bfloat16:  # numpy supports all dtypes torch does except bfloat16
```

Comment:
Instead of comparing with NumPy you can also test bfloat16 by ensuring that the result is always equal to a True tensor of the same shape.